ovs/lib/albanode.py

```diff
@@ -61,33 +133,36 @@ def register(node_id=None, node_type=None, name=None):
         :return: None
         :rtype: NoneType
         """
+        # Generic is a special case. Nothing is registered within config mgmt
         if node_type == AlbaNode.NODE_TYPES.GENERIC:
             node = AlbaNode()
             node.name = name
             node.node_id = ''.join(random.choice(string.ascii_letters + string.digits) for _ in range(32))
             node.type = AlbaNode.NODE_TYPES.GENERIC
             node.save()
         else:
+            # Both S3 and ASD type can be added now
             if node_id is None:
-                raise RuntimeError('A node_id must be given for type ASD')
+                raise RuntimeError('A node_id must be given for type ASD/S3')
             node = AlbaNodeList.get_albanode_by_node_id(node_id)
-            if node is None:
-                main_config = Configuration.get('/ovs/alba/asdnodes/{0}/config/main'.format(node_id))
-                node = AlbaNode()
-                node.name = name
-                node.ip = main_config['ip']
-                node.port = main_config['port']
-                node.username = main_config['username']
-                node.password = main_config['password']
-                node.storagerouter = StorageRouterList.get_by_ip(main_config['ip'])
+            if not node:
+                # Could be a node that was discovered
+                nodes = AlbaNodeController.discover_nodes()
+                for alba_node_guid, alba_node in nodes.iteritems():
+                    if alba_node.node_id == node_id:
+                        node = alba_node
+            if not node:
```

Comment:
twice 
``` if not node```. 
intended or not?

Reply:
Yes, there are 3 paths to take:
- AlbaNode has already been registered -> get that one
- AlbaNode is not registered -> might be coming in because the user wishes to register one of the discovered nodes
- AlbaNode is not registered and the node is not discoverable -> the user is requesting to add something that does not exist

Reply:
I could offload that loop to a new function and raise if it does not find the node to make it more clean..